MTOT.py

```diff
@@ -0,0 +1,260 @@
+            last_id += 1
+            cubit.cmd("delete vol " + str(last_id-1))
+        elif ib_ob == "outboard":
+            cubit.cmd("subtract vol " + str(cyl_id) + " from vol " + str(last_id) + " keep_tool")       
+        cubit.cmd("group 'mat:" + layer_name + "' add vol " + str(last_id))
+
+    return last_id
+
+
+def build_torus(major_radius, minor_radius, radial_build, graveyard):
+    """
+    Generates a circular tokamak geometry in Cubit/Trelis according to a radial build of
+    homogeneous material layers.
+
+    Parameters
+    ----------
+    major_radius : float
+        The plasma major radius
+    minor_radius : float
+        The plasma minor radius
+    radial_build : dict
+        A Python dictionary storing the tokamak radial build data
+    graveyard : bool, optional
+        A flag used to determine whether a graveyard volume is to be added to the geometry
+    """
+    cubit.init([''])
+    vol_id = 0
+    
+    if ('Inboard' in radial_build) or ('Outboard' in radial_build):
+        cubit.cylinder(4*major_radius, major_radius, major_radius, major_radius)
+        vol_id +=1
+        cyl_id = vol_id
+
+    cubit.torus(major_radius, minor_radius)
+    vol_id +=1
+    plasma_id = vol_id
+    cubit.cmd("group 'mat:Vacuum' add vol " + str(plasma_id))
+
+    if 'Inboard' in radial_build:
+        
+        inboard_layers = []
+        inboard_thicknesses = []
+        for index in range(len(radial_build['Inboard'])):
+            pair = radial_build['Inboard'][index]
+            inboard_layers += list(pair.keys())
+            inboard_thicknesses += list(pair.values())
+        
+        contor(major_radius, minor_radius, inboard_thicknesses)
+        vol_id += len(inboard_layers)
+        vol_id = assemble_layers(vol_id, plasma_id, inboard_layers, cyl_id, "inboard")
+
+        first_outboard_id = vol_id + 1
+        
+        outboard_layers = []
+        outboard_thicknesses = []
+        for index in range(len(radial_build['Outboard'])):
+            pair = radial_build['Outboard'][index]
+            outboard_layers += list(pair.keys())
+            outboard_thicknesses += list(pair.values())
+
+        contor(major_radius, minor_radius, outboard_thicknesses)
+        vol_id += len(outboard_layers)
+        vol_id = assemble_layers(vol_id, first_outboard_id, outboard_layers, cyl_id, "outboard")
+
+        cubit.cmd("subtract volume " + str(plasma_id) + " from volume " + str(first_outboard_id) + " keep_tool")
+        vol_id += 1
+        cubit.cmd("subtract vol " + str(cyl_id) + " from vol " + str(vol_id) + " keep_tool")
+        cubit.cmd("group 'mat:" + outboard_layers[0] + "' add vol " + str(vol_id))
+        
+        cubit.cmd("delete vol " + str(cyl_id))
+
+    else:
+        layers = []
+        thicknesses = []
+        for index in range(len(radial_build)):
+            pair = radial_build[index]
+            layers += list(pair.keys())
+            thicknesses += list(pair.values())
```

Comment:
see above for inboard

Probably should be a function to unpack layers & thicknesses and reuse it in all three cases